akd_core/src/verify/history.rs

```diff
@@ -193,6 +222,208 @@ pub fn key_history_verify<TC: Configuration>(
     Ok(results)
 }
 
+fn verify_with_history_params(
+    current_epoch: u64,
+    akd_label: &AkdLabel,
+    proof: &HistoryProofV2,
+    params: HistoryParams,
+) -> Result<(Vec<u64>, Vec<u64>), VerificationError> {
+    let num_proofs = proof.update_proofs.len();
+
+    // Make sure the update proofs are non-empty
+    if num_proofs == 0 {
+        return Err(VerificationError::HistoryProof(format!(
+            "No update proofs included in the proof of user {akd_label:?} at epoch {current_epoch:?}!"
+        )));
+    }
+
+    // Check that the sent proofs are for a contiguous sequence of decreasing versions
+    for count in 0..num_proofs {
+        if count > 0 {
+            // Make sure this proof is for a version 1 more than the previous one.
+            if proof.update_proofs[count].version + 1 != proof.update_proofs[count - 1].version {
+                return Err(VerificationError::HistoryProof(format!(
+                    "Update proofs should be ordered consecutively and in decreasing order.
+                    Error detected with version {} = {}, followed by version {} = {}",
+                    count,
+                    proof.update_proofs[count].version,
+                    count - 1,
+                    proof.update_proofs[count - 1].version
+                )));
+            }
```

Comment:
Since we're using the versions in a few spots (if-expression check and logging), I think we can just move these to variables to clean this up a bit:

```rust
            let prev_version = proof.update_proofs[count - 1].version;
            let curr_version = proof.update_proofs[count].version;

            if curr_version + 1 != prev_version {
                return Err(VerificationError::HistoryProof(format!(
                    "Update proofs should be ordered consecutively and in decreasing order.
                    Error detected with version {count} = {curr_version}, followed by version {} = {prev_version}",
                    count - 1,
                )));
            }
```

Reply:
Sounds good! 

on the error message, how about this "Error detected with version {curr_version} at index {count}, followed by version {prev_version} at index {count-1}"?

Reply:
I updated the message. Let me know what you think!